pkg/dist/buildmodule.go

```diff
@@ -59,6 +61,30 @@ type Target struct {
 	Distributions []Distribution `json:"distributions,omitempty" toml:"distributions,omitempty"`
 }
 
+// ValuesAsSlide converts the internal representation of a target, os, arch, variant, etc. into a string slide
+// each value included into the final array must be not empty.
+func (t *Target) ValuesAsSlide(version string) []string {
+	var targets []string
+	if t.OS != "" {
+		targets = append(targets, t.OS)
+	}
+	if t.Arch != "" {
+		targets = append(targets, t.Arch)
+	}
+	if t.ArchVariant != "" {
+		targets = append(targets, t.ArchVariant)
+	}
+
+	for _, d := range t.Distributions {
+		for _, v := range d.Versions {
```

Comment:
Similar comment re: above... and also, what if distro is non-empty and version is empty? Do we need to omit the `@`?